src/pn532.cpp

```diff
@@ -0,0 +1,296 @@
+/*
+ * Author: Matthias Akstaller
+ */
+
+#if defined(ENABLE_DEBUG) && !defined(ENABLE_DEBUG_NFCREADER)
+#undef ENABLE_DEBUG
+#endif
+
+#include "pn532.h"
+#include <Wire.h>
+#include "app_config.h"
+#include "debug.h"
+#include "lcd.h"
+
+#ifndef I2C_SDA
+#define I2C_SDA 21
+#endif
+
+#ifndef I2C_SCL
+#define I2C_SCL 22
+#endif
+
+#define  SCAN_FREQ            1000
+
+#define MAXIMUM_UNRESPONSIVE_TIME  60000UL //after this period the pn532 is considered offline
+#define AUTO_REFRESH_CONNECTION         30 //after this number of polls, the connection to the PN532 will be refreshed
+
+
+/*
+ * Documentation of the I2C messages:
+ * NXP PN532 manual, https://www.nxp.com/docs/en/user-guide/141520.pdf
+ * (accessed: 2022-01-14)
+ */
+
+#define PN532_I2C_ADDRESS  (0x48 >> 1)
+#define PN532_RDY 0x01
+#define PN532_FRAME_HEADER_LEN 4
+#define PN532_FRAME_IDENTIFIER_LEN 1
+#define PN532_CMD_SAMCONFIGURATION_RESPONSE 0x15
+#define PN532_CMD_INAUTOPOLL_RESPONSE 0x61
+
+#define I2C_READ_BUFFSIZE 35 //might increase later
+
+PN532::PN532() : MicroTasks::Task() {
+    
+}
+
+void PN532::begin() {
+    Wire.begin(I2C_SDA, I2C_SCL);
+    MicroTask.startTask(this);
+}
+
+unsigned long PN532::loop(MicroTasks::WakeReason reason){
+
+    read();
+
+    if (!config_rfid_enabled()) {
+        status = DeviceStatus::NOT_ACTIVE;
+        return SCAN_FREQ;
```

Comment:
No need to poll if RFID is not enabled, can just return `MicroTask.Infinate` and manually wake the task when the config is changed.

Reply:
Okay. Only for the PN532 driver or should I also do it so in the `RfidTask`? And what would be the best way to detect when `config_rfid_enabled()` turns on again? (another case as example is sufficient)